### PR TITLE
feat: eth liquidation

### PR DESCRIPTION
**Pull request by GopherJ**
Signed-off-by: Cheng JIANG <alex_cj96@foxmail.com>

## Security Checklist

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

File: contracts/protocol/libraries/logic/LiquidationLogic.sol [hunk 1]
```diff
@@ -594,12 +617,12 @@ library LiquidationLogic {
 
         uint256 maxLiquidatableDebt = userDebt.percentMul(closeFactor);
 
-        uint256 actualDebtToLiquidate = params.liquidationAmount >
+        uint256 actualLiquidationAmount = params.liquidationAmount >
```
Comment (yz89) [inline contracts/protocol/libraries/logic/LiquidationLogic.sol:620]: lets use Math.min

File: contracts/protocol/libraries/logic/LiquidationLogic.sol [hunk 2]
```diff
@@ -672,7 +695,7 @@ library LiquidationLogic {
             .getLiquidationProtocolFee();
 
         uint256 maxCollateralToLiquidate = ((vars.liquidationAssetPrice *
```
Comment (yz89) [inline contracts/protocol/libraries/logic/LiquidationLogic.sol:698]: I think `maxCollateralToGet` is better.

**Review by zhoujia6139 — APPROVED** [review]
Looks good to me

File: contracts/protocol/libraries/logic/LiquidationLogic.sol [hunk 3]
```diff
@@ -831,4 +855,27 @@ library LiquidationLogic {
             );
         }
     }
+
+    /**
+     * @notice Convert msg.value to WETH and check if liquidationAsset is WETH (if msg.value > 0)
```
Comment (yz89) [inline contracts/protocol/libraries/logic/LiquidationLogic.sol:860]: should it means: Convert msg.value to WETH and check if liquidationAsset is `ETH` (if msg.value > 0) ?

**Pull request merged**